builder/simplify_exprs.cc

```diff
@@ -690,10 +690,9 @@ expr simplify(const less* op, expr a, expr b) {
 
       // Nested logicals
       r.rewrite(x < y, y && !x, is_logical(x) && is_logical(y)) ||
-      r.rewrite(x < (y + c0), true, is_logical(x) && is_logical(y) && eval(c1 > 1)) ||
-      r.rewrite(x < (y + c0), false, is_logical(x) && is_logical(y) && eval(c1 < -1)) ||
-      r.rewrite((x + c0) < y, false, is_logical(x) && is_logical(y) && eval(c1 >= 1)) ||
-      r.rewrite((x + c0) < y, true, is_logical(x) && is_logical(y) && eval(c1 <= -1)) ||
+      r.rewrite(x < 1, !x, is_logical(x)) ||
+      r.rewrite(x < c0, true, eval(c0 > 1) && is_logical(x)) ||
+      r.rewrite(x < c0, false, eval(c0 <= 0) && is_logical(x)) ||
```

Comment:
I think we do need the constant on the LHS versions of these rules, because the constant won't end up on the RHS if the LHS would be zero.